spec/services/npq/funding_eligibility_spec.rb

```diff
@@ -3,167 +3,417 @@
+            end
+
+            it "returns falsey for previously_received_targeted_funding_support" do
+              expect(subject.call[:previously_received_targeted_funding_support]).to be_falsey
+            end
+          end
+
+          context "that is previously funded" do
+            before do
+              get_an_identity_application = create(
+                :npq_application,
+                user:,
+                eligible_for_funding: true,
+                teacher_reference_number_verified: true,
+                targeted_delivery_funding_eligibility: true,
+                npq_course:,
+              )
+              NPQ::Application::Accept.new(npq_application: get_an_identity_application).call
+            end
+
+            it "returns truthy for previously_funded" do
+              expect(subject.call[:previously_funded]).to be_truthy
+            end
+
+            it "returns truthy for previously_received_targeted_funding_support" do
+              expect(subject.call[:previously_received_targeted_funding_support]).to be_truthy
+            end
+          end
+        end
       end
 
-      it "returns truthy" do
-        expect(subject.call[:previously_funded]).to be_truthy
-        expect(subject.call[:previously_received_targeted_funding_support]).to be_falsey
+      context "when previously funded" do
+        let(:trn) { application.teacher_reference_number }
+        let(:application) do
+          create(
+            :npq_application,
+            eligible_for_funding: true,
+            teacher_reference_number_verified: true,
+          )
+        end
+        let(:npq_course) { application.npq_course }
+
+        before do
+          NPQ::Application::Accept.new(npq_application: application).call
+        end
+
+        it "returns truthy for previously_funded" do
+          expect(subject.call[:previously_funded]).to be_truthy
+        end
+
+        it "returns falsey for previously_received_targeted_funding_support" do
+          expect(subject.call[:previously_received_targeted_funding_support]).to be_falsey
+        end
       end
-    end
 
-    context "when previously funded for a different course" do
-      let(:trn) { application.teacher_reference_number }
-      let(:application) do
-        create(
-          :npq_application,
-          eligible_for_funding: true,
-          teacher_reference_number_verified: true,
-        )
+      context "when previously funded for a different course" do
+        let(:trn) { application.teacher_reference_number }
+        let(:application) do
+          create(
+            :npq_application,
+            eligible_for_funding: true,
+            teacher_reference_number_verified: true,
+          )
+        end
+        let(:npq_application_course) { application.npq_course }
+        # Making sure they are completely separate courses
+        let(:npq_course) { create(:npq_course, identifier: npq_application_course.identifier.reverse) }
```

Comment:
Id only be worried here about doing a random string rather than a course that exists if we have some weird checking against certain course in future and this test doesnt catch it if makes sense? might be worth just selecting two tests we know exist and comparing them against each other?

Reply:
Updated